tools/ci_build/github/azure-pipelines/mac-react-native-ci-pipeline.yml

```diff
@@ -15,7 +15,7 @@ variables:
     # 'dev' will skip generating onnxruntime-common package when it's not updated
     # since react-native e2e always requires both onnxruntime-common and onnxruntime-react-native packages,
     # use 'custom' type mode here.
-    npm_packaging_mode: '-dev.$(Get-Date -Format yyyyMMdd)-$(git rev-parse --short HEAD)'
```

Comment:
what is the reason of not preserving the old version format for @dev build?

Reply:
Currently, 'dev' mode is working as a 'custom' in pack-npm-packages.ps1 since a parameter is not a 'dev'. In this PR, it relies on 'dev' implementation in the script.